src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -77,6 +77,22 @@ impl SstableStreamIterator {
         }
     }
 
+    async fn prune_from_valid_block_iter(&mut self) -> HummockResult<()> {
+        while let Some(block_iter) = self.block_iter.as_mut() {
+            if self
+                .sstable_info
+                .get_table_ids()
+                .binary_search(&FullKey::decode(block_iter.key()).user_key.table_id.table_id)
```

Comment:
After #8512, we can get table_id directly from block_iterator. Let's wait for that before merging this PR

Reply:
fyi, #8512 is merged

Reply:
> fyi, #8512 is merged

Thank you. I have just updated my code.